core/trino-main/src/test/java/io/trino/sql/query/TestFilterInaccessibleColumns.java

```diff
@@ -0,0 +1,246 @@
+        accessControl.deny(privilege(USER, "nation.comment", SELECT_COLUMN));
+        assertThat(assertions.query("SELECT * FROM nation WHERE name = 'FRANCE'"))
+                .matches("VALUES (BIGINT '6', CAST('FRANCE' AS VARCHAR(25)), BIGINT '3')");
+    }
+
+    @Test
+    public void testDescribe()
+    {
+        assertThat(assertions.query("DESCRIBE nation"))
+                .matches(materializedRows -> materializedRows
+                        .getMaterializedRows().stream()
+                        .filter(materializedRow -> materializedRow.getField(0).equals("comment"))
+                        .findFirst()
+                        .isPresent());
+
+        accessControl.deny(privilege(USER, "nation.comment", SELECT_COLUMN));
+        assertThat(assertions.query("DESCRIBE nation"))
+                .matches(materializedRows -> materializedRows
+                        .getMaterializedRows().stream()
+                        .filter(materializedRow -> materializedRow.getField(0).equals("comment"))
+                        .findFirst()
+                        .isEmpty());
+    }
+
+    @Test
+    public void testShowColumns()
+    {
+        assertThat(assertions.query("SHOW COLUMNS FROM nation"))
+                .matches(materializedRows -> materializedRows
+                        .getMaterializedRows().stream()
+                        .filter(materializedRow -> materializedRow.getField(0).equals("comment"))
+                        .findFirst()
+                        .isPresent());
+
+        accessControl.deny(privilege("nation.comment", SELECT_COLUMN));
+        assertThat(assertions.query("SHOW COLUMNS FROM nation"))
+                .matches(materializedRows -> materializedRows
+                        .getMaterializedRows().stream()
+                        .filter(materializedRow -> materializedRow.getField(0).equals("comment"))
+                        .findFirst()
+                        .isEmpty());
+    }
+
+    /**
+     * Test filtering when columns are explicitly specified in SELECT
+     */
+    @Test
+    public void testFilterExplicitSelect()
+    {
+        // Select the columns that are available to us explicitly
+        accessControl.deny(privilege(USER, "nation.comment", SELECT_COLUMN));
+        assertThat(assertions.query("SELECT nationkey, name, regionkey FROM nation WHERE name = 'FRANCE'"))
+                .matches("VALUES (BIGINT '6', CAST('FRANCE' AS VARCHAR(25)), BIGINT '3')");
+
+        // Select all columns explicitly
+        assertThatThrownBy(() -> assertions.query("SELECT nationkey, name, regionkey, comment FROM nation WHERE name = 'FRANCE'"))
+                .hasMessage("Access Denied: Cannot select from columns [nationkey, regionkey, name, comment] in table or view local.tiny.nation");
+    }
+
+    @Test
+    public void testRowFilterOnNotAccessibleColumnKO()
+    {
+        // No filtering baseline
```

Comment:
same, baseline as separate test method. Please apply this to other tests too.

Reply:
ok